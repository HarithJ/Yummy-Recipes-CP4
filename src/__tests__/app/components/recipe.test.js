import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Recipe from '../../../components/recipes/recipe.js'

describe("Recipe component", () => {
  const recipe = shallow(<Recipe ingredients={[]}/>)

  it("renders properly", () => {
    expect(shallowToJson(recipe)).toMatchSnapshot();
  })

  it("displays properly", () => {
    const recipeObject = {
      "title": "Recipe title",
      "ingredients": ["ingredient1", "ingredient2"],
      "directions": "Directions to cook the recipe"
    }

    const recipeDisplayed = shallow(<Recipe {...recipeObject} />)

    expect(recipeDisplayed.find('.recipetitle').text()).toEqual("Recipe title")
    expect(recipeDisplayed.find('.directions').text()).toEqual("Directions to cook the recipe")
  })
})
