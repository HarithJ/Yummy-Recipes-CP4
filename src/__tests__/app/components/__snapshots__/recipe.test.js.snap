// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component renders properly 1`] = `
<div
  className="row justify-content-center recipe"
>
  <div
    className="col-10"
  >
    <hr />
    <div
      className="row"
    >
      <div
        className="col-10"
      >
        <h3
          className="recipetitle"
        />
      </div>
      <EditRecipe
        recipeIngs={Array []}
      />
      <div
        className="col-auto"
      >
        <button
          className="btn btn-outline-danger rounded-circle"
          onClick={[Function]}
        >
          <i
            className="far fa-trash-alt"
          />
        </button>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-3"
      >
        <h5>
          Ingredients
        </h5>
        <ul />
      </div>
      <div
        className="col-sm-9"
      >
        <h5>
          Directions
        </h5>
        <p
          className="directions"
        />
      </div>
    </div>
  </div>
</div>
`;
